specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2021-09-01/clusters.json

```diff
@@ -0,0 +1,643 @@
+          "readOnly": true
+        }
+      }
+    },
+    "Cluster": {
+      "description": "Cluster details.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "System data of Cluster resource",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "x-ms-client-flatten": true,
+          "readOnly": true
+        },
+        "properties": {
+          "description": "Cluster properties.",
+          "$ref": "#/definitions/ClusterProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ClusterPatch": {
+      "description": "Cluster details to update.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "description": "Cluster properties.",
+          "$ref": "#/definitions/ClusterPatchProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ClusterProperties": {
+      "description": "Cluster properties.",
+      "type": "object",
+      "required": [
+        "aadClientId",
+        "aadTenantId"
+      ],
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state.",
+          "type": "string",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Accepted",
+            "Provisioning"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "status": {
+          "description": "Status of the cluster agent.",
+          "type": "string",
+          "enum": [
+            "NotYetRegistered",
+            "ConnectedRecently",
+            "NotConnectedRecently",
+            "Disconnected",
+            "Error"
+          ],
+          "x-ms-enum": {
+            "name": "Status",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "cloudId": {
+          "description": "Unique, immutable resource id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "cloudManagementEndpoint": {
+          "description": "Endpoint configured for management from the Azure portal.",
+          "type": "string"
+        },
+        "aadClientId": {
+          "description": "App id of cluster AAD identity.",
+          "type": "string"
+        },
+        "aadTenantId": {
+          "description": "Tenant id of cluster AAD identity.",
+          "type": "string"
+        },
+        "desiredProperties": {
```

Comment:
>desiredProperties [](http://example.com/codeflow?start=9&length=17)

This API pattern is fine, but I am curious what the distinction is between `desiredProperties` vs other properties of the cluster. On a resource PUT the entire resource payload describes the desired state of the resource, so wouldn't all the properties be "desired properties"?

Reply:
Hi, so `desiredProperties` is assigned from the cloud service while the other reported properties are assigned from the on-prem service. I hope that clarifies.